File: ios/ios/Config/ConfigInfo.m
//
//  ConfigInfo.m
//  ios
//
//  Created by Michael on 2017/9/30.
//  Copyright © 2017年 weexstudy. All rights reserved.
//

#import "ConfigInfo.h"

#if TARGET_IPHONE_SIMULATOR
#define DEMO_HOST @"127.0.0.1"
#else
#define DEMO_HOST @"192.168.2.3"
#endif

@implementation ConfigInfo

+ (ConfigInfo*) sharedInstance {
    static ConfigInfo *sharedObj = nil;
    static dispatch_once_t predicate;
    
    dispatch_once(&predicate, ^{
        NSString *path = [[NSBundle mainBundle] pathForResource:@"env.plist" ofType:nil];
        if(!path){
            @throw [NSException exceptionWithName:@"HsError Environmental" reason:@"Error Environmental no file" userInfo:nil];
        }
        NSDictionary* envDict = [NSDictionary dictionaryWithContentsOfFile:path];
        if (!envDict) {
            @throw [NSException exceptionWithName:@"HsError Environmental" reason:@"Error Environmental parse error" userInfo:nil];
        }
        
        sharedObj = [[ConfigInfo alloc] init];
        
        NSString* releaseTypeStr = [envDict objectForKey:@"ReleaseType"];
        if([@"DEBUG" isEqualToString:releaseTypeStr]){
            sharedObj->_releaseType = ReleaseTypeDebug;
        } else if([@"PUBLISH" isEqualToString:releaseTypeStr]){
            sharedObj->_releaseType = ReleaseTypePublish;
        } else if([@"DEV" isEqualToString:releaseTypeStr]){
            sharedObj->_releaseType = ReleaseTypeDev;
        }
    });
    
    return sharedObj;
}

#pragma mark
#pragma mark api url
-(NSString *) urlApiBase {
    static NSString* url = nil;
    
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        switch (self.releaseType) {
            case ReleaseTypeDev: // 开发环境
                url = [self devApiUrl];
                break;
            case ReleaseTypeDebug: // 测试
                url = [self debugApiUrl];
                break;
            case ReleaseTypePublish: // 生产
                url = [self publishApiUrl];
                break;
            default:
                url = @"";
                break;
        }
    });
    
    return url;
}

-(NSString *)urlAPILogin {
    return [NSString stringWithFormat:@"%@app/gaas/login", [self urlApiBase]];
}

#pragma mark
#pragma mark api 地址
-(NSString *) devApiUrl {
    return [NSString stringWithFormat:@"http://%@:8008/",DEMO_HOST];
}

-(NSString *) debugApiUrl {
    return [NSString stringWithFormat:@"http://%@:8008/",DEMO_HOST];
}

-(NSString *) publishApiUrl {
    return [NSString stringWithFormat:@"http://%@:8008/",DEMO_HOST];
}

#pragma mark
#pragma mark weex url
-(NSString *) urlWeexBase {
    static NSString* url = nil;
    
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        switch (self.releaseType) {
            case ReleaseTypeDev: // 开发环境
                url = [self devWeexUrl];
                break;
            case ReleaseTypeDebug: // 测试
                url = [self debugWeexUrl];
                break;
            case ReleaseTypePublish: // 生成
                url = [self publishWeexUrl];
                break;
            default:
                url = @"";
                break;
        }
    });
    
    return url;
}

-(NSString *) urlWeexRoot {
    NSString *baseUrl = [self urlWeexBase];
    NSString *url = @"";
    // 空 读取本地js 文件
    if ([baseUrl isEqualToString:@""]) {
        url = @"file://";
    } else {
        url = [NSString stringWithFormat:@"%@dist/weex/",[self urlWeexBase]];
    }
    return url;
}

#pragma mark
#pragma mark weex 开发环境 地址
-(NSString *) devWeexUrl {
    return [NSString stringWithFormat:@"http://%@:8008/",DEMO_HOST];
}

#pragma mark weex 测试 地址
-(NSString *) debugWeexUrl {
    return @"";
}

#pragma mark weex 生产环境 地址
-(NSString *) publishWeexUrl {
    return @"";
}

#pragma mark
#pragma mark vue url 分享、事件模板的url地址配置
-(NSString *) urlVueBase {
    static NSString* url = nil;
    
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        switch (self.releaseType) {
            case ReleaseTypeDev: // 开发环境
                url = [self devVueUrl];
                break;
            case ReleaseTypeDebug: // 测试
                url = [self debugVueUrl];
                break;
            case ReleaseTypePublish: // 生成
                url = [self publishVueUrl];
                break;
            default:
                url = @"";
                break;
        }
    });
    
    return url;
}

#pragma mark
#pragma mark vue 开发环境 地址
-(NSString *) devVueUrl {
//    return [NSString stringWithFormat:@"http://10.26.160.192:8990/"];
    return [NSString stringWithFormat:@"http://%@:8081/",DEMO_HOST];
}

#pragma mark vue 测试 地址
-(NSString *) debugVueUrl {
    return [NSString stringWithFormat:@"http://%@:8081/",DEMO_HOST];
}

#pragma mark vue 生产环境 地址
-(NSString *) publishVueUrl {
    return [NSString stringWithFormat:@"http://%@:8081/",DEMO_HOST];
}

@end

File: Android/app/src/main/java/com/softbi/config/ConfigInfo.java
package com.softbi.config;

import com.softbi.BuildConfig;

/**
 * Created by michael on 2017/10/21.
 */

public class ConfigInfo {
    private static String RELEASE_TYPE = BuildConfig.releaseType;

    public  enum ReleaseType {
        DEBUG,        //测试
        PUBLISH,     //生产
        DEV   //开发环境
    }

    private static ConfigInfo singleInstance = null;
    private ReleaseType releaseType;
    private String baseUrl;

    private void init() {
        ReleaseType releaseType = ReleaseType.valueOf(RELEASE_TYPE);
        this.releaseType = releaseType;
    }

    public static ConfigInfo getInstance() {
        if (singleInstance != null) {
            return singleInstance;

        }
        synchronized (ConfigInfo.class) {
            if (singleInstance != null) {
                return singleInstance;
            }
            ConfigInfo instance = new ConfigInfo();
            instance.init();

            singleInstance = instance;
        }
        return singleInstance;
    }

    public ReleaseType getReleaseType() {
        return releaseType;
    }

    /**
     * api url
     * @return
     */
    public String getApiBaseUrl() {
        switch (releaseType) {
            case DEBUG:
                this.baseUrl = testApiUrl();
                break;
            case PUBLISH:
                this.baseUrl = publishApiUrl();
                break;
            case DEV:
                this.baseUrl = devApiUrl();
                break;
            default:
                this.baseUrl = devApiUrl();
                break;
        }
        return this.baseUrl;
    }
    public String getAPILoginUrl() {
        return this.getApiBaseUrl() + "app/gaas/login";
    }

    public String getLocalApiUrl () {
//        return "http://10.26.160.192:8031/front/";
        return "http://36.110.68.73:8082/gaas/";
    }
    /**
     * api 开发环境
     */
    private String devApiUrl() {
        return "http://im.one2rich.cn/gaas/";
//        return "http://10.26.160.192:8030/gaas/";
//        return "http://10.26.160.187:8082/gaas/";
//        return "http://10.26.160.193:8080/gaas/";
//        return "http://10.26.160.192:8030/front/";
    }
    /**
     * api 测试环境
     */
    private String testApiUrl() {
        return "http://im.one2rich.cn/gaas/";
//        return "http://10.26.160.192:8030/gaas/";
    }
    /**
     * api 生产环境
     */
    private String publishApiUrl() {
        return "http://im.one2rich.cn/gaas/";
        //return "http://47.96.146.178/gaas/";
    }

    /**
     * Weex url
     * @return
     */
    public String getWeexBaseUrl() {
        switch (releaseType) {
            case DEBUG:
                this.baseUrl = testWeexUrl();
                break;
            case PUBLISH:
                this.baseUrl = publishWeexUrl();
                break;
            case DEV:
                this.baseUrl = devWeexUrl();
                break;
            default:
                this.baseUrl = devWeexUrl();
                break;
        }
        return this.baseUrl;
    }
    /**
     * weex 首页连接
     */
    public String getWeexHomeUrl() {
        return this.getWeexBaseUrl() + "dist/weex/index.js";
    }
    /**
     * weex 获取根路径 url
     */
    public String getWeexRootUrl() {
        String baseUrl = this.getWeexBaseUrl();
        if ("".equals(baseUrl)) {
            return "file:///weex/";
        } else {
            return baseUrl + "dist/weex/";
        }
    }
    /**
     * weex 开发环境
     */
    private String devWeexUrl() {;
         return "http://192.168.2.3:8008/";
    }
    /**
     * weex 测试环境
     */
    private String testWeexUrl() {
        return "";
    }
    /**
     * weex 生产环境
     */
    private String publishWeexUrl() {
        return "";
    }

    /**
     * vue url
     * @return
     */
    public String getVueBaseUrl() {
        switch (releaseType) {
            case DEBUG:
                this.baseUrl = testVueUrl();
                break;
            case PUBLISH:
                this.baseUrl = publishVueUrl();
                break;
            case DEV:
                this.baseUrl = devVueUrl();
                break;
            default:
                this.baseUrl = devVueUrl();
                break;
        }
        return this.baseUrl;
    }
    /**
     * vue 开发环境
     */
    private String devVueUrl() {
        return "http://192.168.2.3:8081/";
    }
    /**
     * vue 测试环境
     */
    private String testVueUrl() {
        return "http://im.one2rich.cn:8083/";
//        return "http://192.168.2.3:8081/";
    }
    /**
     * vue 生产环境
     */
    private String publishVueUrl() {
//        return "http://192.168.2.3:8081/";
        return "http://im.one2rich.cn:8083/";
    }

}
